app/graphql/types/query_type.rb

```diff
@@ -1,11 +1,16 @@
 class Types::QueryType < Types::BaseObject
   field :locations, [Types::LocationType] do
     description "MojoTech office locations"
+    argument :city_name, String, "City name of location", required: false
   end
 
-  def locations
-    Rails.cache.fetch('locations-response', expires_in: 15.minutes) do
-      Location.all
+  def locations(city_name: nil)
+    if city_name
```

Comment:
I think this would be more accurately represented as a separate property in the schema `def location(city_name)` where city_name was not optional and we only ever return on location. Ideally, methods do just one thing.